packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1137,21 +1145,45 @@ class RouteObserver<T extends Route<dynamic>> extends NavigatorObserver {
   /// [routeAware] is no longer informed about changes to its route.
```

Comment:
Add a comment saying that if the given argument was subscribed to multiple types, this will unregister it (once) from each type.